### PR TITLE
Refactor: MiniWallet functionality for more readable code in functional test `rpc_signtransactionwithkey.py`

### PR DESCRIPTION
**Pull request by allanperlee**
Addressing issue #30600, this pull request simplifies the `send_to_address` functional test in rpc_signrawtransactionwithkey.py. The code is more terse with a MiniWallet object, its ADDRESS_OP_TRUE mode and `send_to` method on line 54, thus removing the "low-level" details of the transaction. Suggestions will be addressed. Thank you for reviewing!

### CODE REVIEW

**Review by ismaelsadeeq — COMMENTED** [review]
Thank you for attempting to address this issue. 

The changes in this PR are incorrect and do not address the issue. 
Therefore, I am inclined to approach NACK. 
If you have any questions about this PR, please ask them in the issue and review the guidelines at [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) before opening a new PR.

Hint to anyone wanting to work on this.
check `MiniWallet::send_to` method.
https://github.com/bitcoin/bitcoin/blob/ee367170cb2acf82b6ff8e0ccdbc1cce09730662/test/functional/test_framework/wallet.py#L273-L294

And convert the address in `SignRawTransactionWithKeyTest::send_to_address` to a scriptPubKey before passing it to `MiniWallet::send_to` method.

**Comment by allanperlee** [comment]
@ismaelsadeeq Thanks for reviewing the PR. I corrected the mistake. I really want to get this right, I really appreciate the suggestions.

File: test/functional/rpc_signrawtransactionwithkey.py [hunk 1]
```diff
@@ -48,13 +50,12 @@ class SignRawTransactionWithKeyTest(BitcoinTestFramework):
...
-
+    
     def send_to_address(self, addr, amount):
-        input = {"txid": self.nodes[0].getblock(self.block_hash[self.blk_idx])["tx"][0], "vout": 0}
-        output = {addr: amount}
-        self.blk_idx += 1
-        rawtx = self.nodes[0].createrawtransaction([input], output)
-        txid = self.nodes[0].sendrawtransaction(self.nodes[0].signrawtransactionwithkey(rawtx, [self.nodes[0].get_deterministic_priv_key().key])["hex"], 0)
+        script_pub_key = address_to_scriptpubkey(addr).hex()
+        wallet = MiniWallet(self.node[0], mode=MiniWalletMode.ADDRESS_OP_TRUE)
```
[the remark this answers is not in the document]
Comment (allanperlee) [reply test/functional/rpc_signrawtransactionwithkey.py:56]: I see your point, the issue was readability in this particular test, according to the issuer. Should I leave this alone and write a separate test? `signrawtransactionwithkey` is tested in other parts of this functional test.

**Review by glozow — COMMENTED** [review]
I'm going to close this pull request for now. If you'd like to work on the issue, here are some hints before you open a new pull request.

1. Please ensure your code works before submitting a pull request. There is a lot of documentation for the functional test suite if you need help https://github.com/bitcoin/bitcoin/blob/master/test/functional/README.md
2. Write code a bit more carefully. `node` isn't an attribute; you forgot an `s`. `MiniWallet` does not have a `sendrawtransaction` function; you perhaps confused that with `send_to_address`. The new wallet also doesn't have any utxos to spend; it should probably be instantiated at the start of the test and be reused.
3. Maybe create a branch instead of PRing from your master https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell
4. Squash your commits https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

**Pull request closed without merging**